sdk/dapp-kit/src/hooks/wallet/useCurrentWallet.ts

```diff
@@ -7,5 +7,46 @@ import { useWalletStore } from './useWalletStore.js';
  * Retrieves the wallet that is currently connected to the dApp, if one exists.
  */
 export function useCurrentWallet() {
-	return useWalletStore((state) => state.currentWallet);
+	const currentWallet = useWalletStore((state) => state.currentWallet);
+	const connectionStatus = useWalletStore((state) => state.connectionStatus);
+
+	switch (connectionStatus) {
+		case 'connecting':
+			return {
+				connectionStatus,
+				currentWallet: null,
+				isDisconnected: false,
+				isConnecting: true,
+				isReconnecting: false,
+				isConnected: false,
+			} as const;
+		case 'reconnecting':
```

Comment:
I think `reconnecting` is a weird name, because this is really auto connect is happening, but I think this is often used as a term when you already establish a connection and need to re-initiate it due to disconnect. That's _kind of_ what has happened but I don't think most devs will think of it that way.

This also doesn't seem to address my needs, because this will still render in an initial state of not connected, which would trigger an auth redirect. I think we might actually need something like `hasAttemptedAutoConnection` to get knowledge about when auto-connection is complete?

Reply:
re: the first point, I also thought `reconnecting` was weird but then I came back to it because I felt like having the `autoconnecting` terminology in `useConnectWallet` was muddying the hook haha. If we do end up keeping this, I _probably_ won't use the `useConnectWallet` hook for handling auto-connecting because then we'd have to further differentiate between "is an auto-connection attempt happening" vs. "is someone connecting to a wallet account that they were most recently connected to".  but this all might be pointless depending on your next point haha

Okay re: the second point, that makes sense and hmmm this is kind of tricky to solve 🤔. My first inclination would be to say "let's move auto-connecting" outside of the React lifecycle, but that would require creating the wallet provider config outside of React which we might not want to commit to

The `hasAttemptedAutoConnection` var works, it just doesn't feel very elegant haha. We could basically get this for free in react-query v5 by creating a small wrapper over https://tanstack.com/query/v5/docs/react/reference/useMutationState but that won't be for a while

Reply:
Moving autoconnect outside of the React lifecycle doesn't fully solve it, it just probabilistically solves it (connect is async tho, so very well could still be happening during mount).

Maybe we just do the hacky `useAutoConnectStatus` hook for now, and change the underlying implementation in the future? It also maybe should return if autoConnect is even enabled?

Reply:
true good point, yeah that sounds fair 🤷🏼‍♂️. thanks for the input